src/js/components/sidenav.js

```diff
@@ -3,125 +3,121 @@
-       * childMenus[index]: menu list
-       */
-
-      // Since JavaScript is available add popup semantics to toggles
-      element.setAttribute('aria-haspopup', 'true');
-
-      // Aria-expand all toggles if everything is open on init
-      if (this.openAllOnInit) {
-        element.setAttribute('aria-expanded', 'true');
-        childMenus[index].removeAttribute('hidden');
-
-        return;
-      }
-
-      // Check if this toggle been manually set to expanded in markup
-      if (element.getAttribute('aria-expanded') === 'true') {
-        // Open list matching this toggle
-        childMenus[index].removeAttribute('hidden');
-      } else {
-        element.setAttribute('aria-expanded', 'false');
-        childMenus[index].setAttribute('hidden', '');
+  static get methods() {
+    return {
+      init() {
+        console.log('Sidenav::init');
+
+        this.toggleAttribute = 'data-rvt-sidenav-toggle';
+        this.toggleSelector = `[${this.toggleAttribute}]`;
+        this.listAttribute = 'data-rvt-sidenav-list';
+        this.listSelector = `[${this.listAttribute}]`;
+        this.openAllOnInit = this.element.hasAttribute('data-rvt-sidenav-open-all');
+
+        const menuToggles = nodeListToArray(
+          this.element.querySelectorAll(this.toggleSelector)
+        );
+        const childMenus = nodeListToArray(
+          this.element.querySelectorAll(this.listSelector)
+        );
+    
+        menuToggles.forEach((element, index) => {
+          /**
+           * Toggle button/menu list pairings
+           * element: toggle button
+           * childMenus[index]: menu list
+           */
+    
+          // Since JavaScript is available add popup semantics to toggles
+          element.setAttribute('aria-haspopup', 'true');
+    
+          // Aria-expand all toggles if everything is open on init
+          if (this.openAllOnInit) {
+            element.setAttribute('aria-expanded', 'true');
+            childMenus[index].removeAttribute('hidden');
+    
+            return;
+          }
+    
+          // Check if this toggle been manually set to expanded in markup
+          if (element.getAttribute('aria-expanded') === 'true') {
+            // Open list matching this toggle
+            childMenus[index].removeAttribute('hidden');
+          } else {
+            element.setAttribute('aria-expanded', 'false');
+            childMenus[index].setAttribute('hidden', '');
+          }
+        });
+
+        // bind methods
+        Component.bindMethodToDOMElement(this, 'open', this.open);
+        Component.bindMethodToDOMElement(this, 'close', this.close);
+
+        this._handleClick = this._handleClick.bind(this);
+      },
+
+      connected() {
+        Component.dispatchComponentAddedEvent(this.element);
+      },
+
+      disconnected() {
+        Component.dispatchComponentRemovedEvent(this.element);
+      },
+
+      onClick(event) {
+        this._handleClick(event);
+      },
+
+      _handleClick(event) {
```

Comment:
Similar to some of the other event listener comments. Since this method only ever really gets used in set up and tear down, I think it makes sense to just move the body of the `_handleClick(event)` function into a wicked-element `onclick(event)` method so we get the setup and tear down for free.

I'll write this up a separate refactoring pass though.